src/appleseed/renderer/modeling/aov/cryptomatte.cpp

```diff
@@ -639,8 +661,8 @@ namespace
                 {
                     const float weight = m_renderer_filter->evaluate(rx - dx, ry - dy);
                     WeightMap& weight_map = m_pixel_samples[(m_tile_origin_y + ry) * m_frame_width + (m_tile_origin_x + rx)];
-                    const float old_value = weight_map.get(m3hash);
-                    weight_map.insert(m3hash, old_value + weight); // possible race conditions writing to pixels on the tile edges
+                    float* APPLESEED_RESTRICT ptr = weight_map.get_value_ptr(m3hash);
```

Comment:
`get_value_ptr()` is not thread-safe, is that OK?

Reply:
I guess it's not OK.
It's probably that it creates new entries in the dictionary, is that what makes it unsafe, right?

Reply:
Yes.

Reply:
Hi @dictoon, is it possible to make thread safe dictionary without guards/mutexes? I think it should be something quick and simple right? Shouldn't block the thread I think..
Maybe it's ok to have thread collisions on some of the samples at the edge of the tiles?

Reply:
Forgot to mention - I checked arnold's cryptomatte and it looks like they don't do thread safe writes on their sample buffer. It's possible they have totally different implementation of the tiles but I was just curious.

Reply:
Hi @dictoon , I thought maybe it should have a sample storage per tile and that will make it thread safe? Sorry it takes me long time to fix properly.

Reply:
I think I misunderstood and tile doesn't write outside its boundaries, so maybe it was perfectly thread safe before.

Reply:
That's correct.